src/languages/gml.js

```diff
@@ -2994,135 +937,572 @@ export default function(hljs) {
+    className: "literal",
+    variants: [
+      { match: /(\B|^)\$[0-9a-fA-F]+/ },
+      { match: /(\B|^)#[0-9a-fA-F]+/ },
+      { match: /\b0x[0-9a-fA-F][0-9a-fA-F_]*/ },
+      { match: /\b0b[01][01_]*/ },
+      { match: /\b[0-9][0-9_.]*/ }
+    ]
+  };
+
+  const COMMENT_LINE_INNER = {
+    match: /[^\n]+/,
+    className: "comment"
+  };
+
+  /**
+   * Pre-processor modes for macro definitions and regions.
+   */
+  const PREPROCESSOR = {
+    end: /$/,
+    variants: [
+      {
+        begin: [
+          /#macro\s+/,
+          VALID_IDENTIFIER_REG
+        ],
+        className: {
+          1: "keyword",
+          2: "literal"
+        },
+        contains: lateConcat(
+          EXPRESSION,
+          [
+            COMMENT,
+            {
+              match: /\\\n/
+            }
+          ]
+        ),
+      },
+      {
+        begin: "#define"
+      },
+      {
+        begin: /#(end)?region\b/,
+        beginScope: "keyword",
+        contains: [COMMENT_LINE_INNER]
+      },
+    ]
+  };
+
+  /**
+   * Dot accessor usage with a special highlighting case for `global`.
+   */
+  const PROP_ACCESS = [
+    {
+      match: [
+        "global",
+        DOT_ACCESSOR_REG,
+        VALID_IDENTIFIER_REG
+      ],
+      className: {
+        1: "literal",
+        3: "property"
+      }
+    },
+    {
+      match: [
+        DOT_ACCESSOR_REG,
+        VALID_IDENTIFIER_REG,
+        /\s*\(/
+      ],
+      className: {
+        2: "function"
+      }
+    },
+    {
+      match: [
+        DOT_ACCESSOR_REG,
+        VALID_IDENTIFIER_REG
+      ],
+      className: {
+        2: "property"
+      }
+    },
+  ];
+
+  /**
+   * Function call sites, just looking for `<ident>(`. This creates false positives
+   * for keywords such as `if (<condition>)`, so has lower priority in the mode `contains` list.
+   */
+  const FUNCTION_CALL = {
+    begin: [
+      VALID_IDENTIFIER_REG,
+      /\s*?/,
+      /\(/
+    ],
+    className: {
+      1: "function"
+    }
+  };
+
+  /**
+   * The manual likes using `obj_` and such to define assets. Sneaky trick to make it look nicer :P
+   */
+  const USER_ASSET_CONSTANT = {
+    className: "literal",
+    end: VALID_IDENTIFIER_REG,
+    variants: [
+      { begin: "spr_" },
+      { begin: "obj_" },
+    ]
+  };
+
+  /**
+   * A ternary expression, matching partial ternary as `? <EXPRESSION> :`.
+   * Effectively exists to prevent {@link STRUCT_LITERAL_MEMBER} from stealing `<EXPRESSION> :`.
+   */
+  const TERNARY = {
+    begin: /\?/,
+    end: /:/,
+    contains: EXPRESSION
+  };
+
+  const SWITCH_CASE = {
+    begin: [
+      /case/,
+      /\s+/
+    ],
+    end: /:/,
+    className: {
+      1: "keyword"
+    },
+    contains: EXPRESSION
+  };
+  
+  /**
+   * A struct variable declaration, of `<ident>:`
+   */
+  const STRUCT_LITERAL_MEMBER = {
+    match: [
+      /\b/,
+      VALID_IDENTIFIER_REG,
+      /\s*:/
+    ],
+    className: {
+      2: "variable-instance"
```

Comment:
This is not one of the scopes we support.

Reply:
Leftover from some fine-grained extra scopes I had in the manual.